api/action/index.js

```diff
@@ -1,14 +1,27 @@
-const EMOJI = [
-  'confetti',
-  'suit',
-  'bow',
-  'dog',
-  'skull',
-  'yoyo',
-  'cat',
-];
+/**
```

Comment:
Sorry to rewrite your code but this should look like:

```js
/**
 * @param {number} timeout
 * @param {(event: Event) => void} callback
 * @return {(event: Event) => void}
 */
function debounce(timeout, callback) {
  let timeoutID = 0;
  return (event) => {
    clearTimeout(timeoutID);
    timeoutID = setTimeout(() => callback(event), timeout);
  };
}
```

The convenient thing here is that you can safely call `clearTimeout` on a zero ID, or an ID which has already fired. It's harmless.

Reply:
Also [MDN agrees](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout) that it's safe to always call `clearTimeout` on an oldID:

> It is guaranteed that a timeout ID will never be reused by a subsequent call to setTimeout() or setInterval() on the same object (a window or a worker).

TIL you can probably only ever have [MAX_SAFE_INTEGER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER) calls to `setTimeout`.

Reply:
I'm glad for the rewrite. Mine was functional but it felt like it was more complicated than the other debug functions I vaguely recalled. I'll integrate this shortly.